core/src/main/java/com/sequenceiq/cloudbreak/service/multiaz/DataLakeMultiAzCalculatorService.java

```diff
@@ -0,0 +1,91 @@
+package com.sequenceiq.cloudbreak.service.multiaz;
+
+import static org.apache.commons.lang3.StringUtils.containsIgnoreCase;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceGroup;
+import com.sequenceiq.cloudbreak.domain.stack.instance.InstanceMetaData;
+import com.sequenceiq.cloudbreak.domain.stack.instance.network.InstanceGroupNetwork;
+import com.sequenceiq.common.api.type.InstanceGroupName;
+
+@Service
+public class DataLakeMultiAzCalculatorService extends MultiAzCalculatorService {
+
+    private static final String SDX_ENTERPRISE_DATALAKE_TEXT = "Enterprise SDX";
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(DataLakeMultiAzCalculatorService.class);
+
+    @Override
+    public void calculateByRoundRobin(Map<String, String> subnetAzPairs, Stack stack) {
+        if (!containsIgnoreCase(stack.getCluster().getBlueprint().getBlueprintText(), SDX_ENTERPRISE_DATALAKE_TEXT)) {
+            super.calculateByRoundRobin(subnetAzPairs, stack);
+        } else {
+            String stackSubnetId = getStackSubnetIdIfExists(stack);
+            String stackAz = stackSubnetId == null ? null : subnetAzPairs.get(stackSubnetId);
+            datalakeMultiAZCalculator(stack, subnetAzPairs, stackSubnetId, stackAz);
+        }
+    }
+
+    public void datalakeMultiAZCalculator(Stack stack, Map<String, String> subnetAzPairs, String stackSubnetId, String stackAz) {
```

Comment:
This method could be refactored into multiple private methods instead of having this huge one.

Reply:
@lacikaaa addressed your comment.